src/auction/Auction.sol

```diff
@@ -82,9 +75,11 @@ contract Auction is IAuction, VersionedContract, UUPS, Ownable, ReentrancyGuard,
         // Store DAO's ERC-721 token
         token = Token(_token);
 
+        AuctionParams memory params = abi.decode(_data, (AuctionParams));
```

Comment:
This also could be an inner struct – depends on how much you want to change things.

Reply:
@neokry